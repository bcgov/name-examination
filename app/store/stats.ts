import { getStats } from '~/util/namex-api'
import { getFormattedDateWithTime } from '~/util/date'
import { sortNameChoices } from '~/util'
import { type NameRequest } from '~/types'
import { defineStore } from 'pinia'

export const useStatsStore = defineStore('stats', () => {
  /**
   * Fetches paged statistics data based on the provided timespan and myStats flag.
   *
   * @param {string | null} timespan - The timespan for the statistics data. If null, the default timespan is used.
   * @param {boolean} myStats - A flag indicating whether to fetch statistics related to user's own stats.
   * @return {Promise<any>} A promise that resolves with the fetched statistics data.
   * @throws {Error} Throws an error if the fetching process fails.
   */
  async function getPagedStats (timespan: string | null, myStats: boolean): Promise<any> {
    const params = new URLSearchParams({
      timespan: timespan || '1',
      currentpage: '1',
      perpage: '1000'
    })
    if (myStats) params.append('myStats', 'True')

    try {
      const response = await getStats(params)

      if (response.ok) {
        const jsonData = await response.json() // Parse response JSON

        if (jsonData.numRecords >= 0) {
          const requestsData = jsonData.nameRequests as Array<NameRequest>

          const processedData = requestsData.map((statsData:any) => {
            statsData.names = sortNameChoices(statsData.names)
            if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
              statsData.comments[0].timestamp = getFormattedDateWithTime(statsData.comments[0].timestamp)
            }
            statsData.lastUpdate = getFormattedDateWithTime(statsData.lastUpdate)
            statsData.names.forEach((name:any) => {
              if (name.decision_text) {
                name.decision_text = name.decision_text.split(/\n\n/)
              }
            })
            return statsData
          })
          return {
            nameRequests: processedData,
            numRecords: jsonData.numRecords
          }
        }
      } else {
        throw new Error('Network response was not ok')
      }
    } catch (error) {
      console.error('There was a problem with your fetch operation:', error)
      throw error // Rethrow the error to handle it elsewhere if needed
    }
  }

  return {
    getPagedStats
  }
})
