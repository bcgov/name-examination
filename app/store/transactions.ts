import {
  ConsentFlag,
  type EntityTypeCode,
  type RequestActionCode,
  type RequestTypeCode,
} from '~/enums/codes'
import { Status } from '~/enums/nr-status'
import {
  type NameChoice,
  type NameRequest,
  type TransactionEntry,
  type Transactions,
} from '~/types'
import { emitter } from '~/util/emitter'
import { getNameRequest, getTransactions } from '~/util/namex-api'
import { toMappedRequestType } from '~/util/request-type'
import requestTypes from '~/data/request_types.json'
import { sortNameChoices } from '~/util'

export const useTransactions = defineStore('transactions', () => {
  const nr = ref<NameRequest>()
  const transactions = ref<Array<TransactionEntry>>([])
  const loadingNr = ref(false)
  const loadingTransactions = ref(false)

  async function loadNr(nrNumber: string) {
    loadingNr.value = true
    const nrResponse = await getNameRequest(nrNumber)
    nr.value = await nrResponse.json()
    loadingNr.value = false
  }

  async function loadTransactions(nrNumber: string) {
    loadingTransactions.value = true
    const transactionsResponse = await getTransactions(nrNumber)
    const transactionsJson = (await transactionsResponse.json()) as Transactions
    transactionsJson.transactions.forEach(t => sortNameChoices(t.names))
    transactions.value = transactionsJson.transactions
    loadingTransactions.value = false
  }

  async function initialize(nrNumber: string) {
    try {
      await loadNr(nrNumber)
      await loadTransactions(nrNumber)
    } catch (e) {
      emitter.emit('error', {
        title: 'Transactions Error',
        message: 'Failed to retrieve transactions data.',
      })
    } finally {
      loadingNr.value = false
      loadingTransactions.value = false
    }
  }

  function getStatusDisplay(status: Status | null, names: Array<NameChoice>) {
    if (!status) return 'N/A'

    let displayState = status as string
    if (status === Status.Conditional) {
      displayState = 'CONDITIONAL APPROVED'
    } else if (status === Status.Consumed && names && names.length > 0) {
      const approvedName = names.find((name) =>
        [Status.Approved, Status.Condition].includes(name.state)
      )
      displayState =
        approvedName?.state === Status.Condition
          ? 'CONDITIONAL APPROVED'
          : 'APPROVED'
      displayState += ` / Used for ${approvedName?.corpNum}`
    }
    return displayState
  }

  function getRequestTypeDisplay(
    requestType: RequestTypeCode,
    action: RequestActionCode
  ) {
    if (!nr.value) return 'N/A'
    const mapped = toMappedRequestType(
      requestType,
      action,
      nr.value.entity_type_cd
    )
    return requestTypes.find((r) => r.value == mapped)?.text || 'N/A'
  }

  function getConsentDisplay(date: string | null, consentFlag: ConsentFlag) {
    return date
      ? 'Required. Received.'
      : consentFlag === ConsentFlag.Required
      ? 'Required. Not yet received.'
      : 'Not required'
  }

  return {
    nr,
    transactions,
    loadingNr,
    loadingTransactions,
    initialize,
    getStatusDisplay,
    getRequestTypeDisplay,
    getConsentDisplay,
  }
})
