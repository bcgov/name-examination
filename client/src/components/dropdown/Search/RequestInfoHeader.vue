/* eslint-disable */
<template>
  <span>
    <div class="row">

      <!-- details col 1 - priority, comments, etc -->
      <div id='div1' class="col-md-5" >
        <div class="row">
          <div class="col-md-3" >
            <span v-if="priority" id="lblPriority" class="ISPRIORITY" >Priority</span>
            <span v-else></span>
          </div>
          <div class="col-md-9">
            <p v-if="!is_editing">{{ requestType }}</p>
            <textarea v-else
                      id="requestType1"
                      v-model="requestType"
                      class="txtArea form-control">
            </textarea>
          </div>
        </div>

        <div class="row">
          <div class="col-md-3" >
            <p class="nr-number" v-bind:class="{ REDnrNum: priority }">{{ nrNumber }}</p>
          </div>
          <div class="col" >
            <p v-if="!is_editing">{{ jurisdiction }}</p>
            <input v-else v-model="jurisdiction" class="form-control" />
          </div>
          <div class="col">
            <p v-if="!is_editing">{{ nuans }}</p>
            <input v-else v-model="nuans" class="form-control" />
          </div>
          <div class="col">
            <p v-if="!is_editing">{{ sk_name }}</p>
            <input v-else v-model="sk_name" class="form-control" />
          </div>
        </div>

        <div class="row">
          <div class="col add-top-padding">
            <div id="internalCommentsDisplay" v-if="!is_editing" style="display: none;">
              <h3>INTERNAL COMMENTS</h3>
              <p>{{ internalComments }}</p>
            </div>
            <div v-else>
              <h3>INTERNAL COMMENTS</h3>
              <textarea v-model="internalComments" class="form-control"></textarea>
            </div>

          </div>
        </div>
      </div>

      <!-- details col 2 - nature of business -->
      <div id='div2' class="col-md-4">
        <h3>NATURE OF BUSINESS</h3>
        <p id="natureOfBusinessDisplay" v-if="!is_editing">{{ natureOfBusiness }}</p>
        <textarea v-else v-model="natureOfBusiness" class="form-control"></textarea>

      </div>

      <!-- details col 3 - additional info, client details -->
      <div id='div3' class="col-md-3">
        <div class="row">
          <div class="col">
            <div class="row">
              <div class="col">
                <h3>ADDITIONAL INFORMATION</h3>
                <p v-if="!is_editing">{{ addInfo }}</p>
                <textarea v-else v-model="addInfo" class="form-control"></textarea>
              </div>
            </div>
            <div class="row">
              <div class="col add-top-padding">
                <clientinfoview />
              </div>
            </div>
          </div>
        </div>
      </div>

    </div>

    <!-- row 2 - buttons -->
    <div class="row">
      <div id='div4' class="col-md-12">
        <span class="f1" @click="toggleDetails">F1</span>
      </div>
      <button v-if="!is_editing" class="btn btn-default" @click="edit">Edit</button>
      <button v-else class="btn btn-default" @click="save">Save</button>
    </div>

  </span>
</template>

<script>
/* eslint-disable */

import clientinfoview from '@/components/dropdown/Search/client/ClientInfo.vue';

export default {
    name: 'RequestInfoHeader',
    computed: {
      is_editing() {
        return  this.$store.getters.is_editing;
      },
      nrNumber() {
        return  this.$store.getters.nrNumber;
      },
      requestType() {
        return  this.$store.getters.requestType;
      },
      firstName() {
        return  this.$store.getters.firstName;
      },
      lastName() {
        return  this.$store.getters.lastName;
      },
      address() {
        return  this.$store.getters.address;
      },
      contactName() {
        return  this.$store.getters.contactName;
      },
      phone() {
        return  this.$store.getters.phone;
      },
      conEmail() {
        return  this.$store.getters.conEmail;
      },
      fax() {
        return  this.$store.getters.fax;
      },
      jurisdiction() {
        return  this.$store.getters.jurisdiction;
      },
      natureOfBusiness() {
        return  this.$store.getters.natureOfBusiness;
      },
      nuans() {
        return  this.$store.getters.nuans;
      },
      sk_name() {
        return  this.$store.getters.sk_name;
      },
      nr_status() {
        return  this.$store.getters.nr_status;
      },
      examiner() {
        return  this.$store.getters.examiner;
      },
      priority() {
        return  this.$store.getters.firstName;
      },
      resubmissionYN() {
        return  this.$store.getters.resubmissionYN;
      },
      linkedNR() {
        return  this.$store.getters.linkedNR;
      },
      reservationCount() {
        return  this.$store.getters.reservationCount;
      },
      expiryDate() {
        return  this.$store.getters.expiryDate;
      },
      details() {
        return this.$store.getters.details;
      },
      internalComments() {
        // TODO get actual data from $store
        return 'Lorem ipsum dolor sit amet, at vix enim nominavi, ea dico case euismod mea, est cu ubique iuvaret. Vis ea rebum verear, salutandi posidonium signiferumque ut qui. Et inermis iudicabit sententiae his, tation facilisis eam id. Adhuc phaedrum inimicus cu sit.\n' +
          '\n' +
          'Ullum melius delicatissimi te sit, ut nullam lucilius rationibus has. Mei sonet moderatius at. Pro legere mentitum omittantur te, natum decore cu pro. Ei brute labores mea, ea assum omittantur nec. Ea per mnesarchum elaboraret necessitatibus, graece nemore id pri. Ut saepe salutatus intellegam sit, magna vivendo nam in, minimum lucilius qui ne. Mei ei quis verear vulputate, cu vis adipisci ullamcorper.\n' +
          '\n' +
          'Sed alterum convenire signiferumque an, quando omittam repudiandae vix id, nec oratio equidem ei. Ea nec tractatos periculis. Dolore pertinacia id mel, pro patrioque forensibus ea. Aperiri virtute sea et.\n' +
          '\n' +
          'Oportere electram in eos, volumus pericula mei cu. Ius suscipit vivendum scripserit in, ne graecis detraxit patrioque nec. Nec laudem aliquid et, hinc vitae nullam est eu. No errem oblique vix, sed no augue novum dicant, ne purto modus phaedrum sed. Inimicus accusamus mei cu. Qui an altera saperet, fierent percipitur ius no, ex mea ignota quaestio.\n' +
          '\n' +
          'Ea dictas omittam nam, in his fugit mediocrem, vim no invidunt indoctum. Veri audire ne eos. Vim everti fierent an, ad usu omnes adolescens honestatis, tempor detraxit at quo. Duo in zril soluta, vim ne ferri periculis theophrastus. Idque utamur quaeque in eam, te eam deleniti periculis.\n';
      }
    },
    components: {
      clientinfoview
    },
    mounted(){
      this.setInterface();
    },
    methods: {
      setInterface(){
        this.setNRNum()
        this.setPriority()
        this.setComp
        this.setNuans()
        this.setSK()
        this.setNOB()
        this.setAddInfo()
      },
      setNRNum() {
      },
      setPriority() {
      },
      setAddInfo() {
      },
      setComp() {
      },
      setNuans() {
      },
      setSK() {
      },
      setNOB() {
      },
      toggleDetails() {
        // KATIE - IN PROGRESS

        // show internal comments
        document.getElementById("internalCommentsDisplay").style.display = '';

        // show complete nature of business, untruncated

        // show client data
        document.getElementsByClassName('RequestInfoHeader')[0].getElementsByClassName('ClientInfo')[0].style.display = 'inline';

      },
      setBorder(id) {
        var tb = document.getElementById(id);
        tb.borderWidth = "1";
      },
      edit() {
        this.$store.state.is_editing = true;
      },
      save() {
        this.$store.state.is_editing = false;
      }
    },
    watch: {
      nrNumber: function (val) {
        this.$store.dispatch('getpostgrescompInfo',this.nrNumber)
        this.setInterface();
      }
    }
}
</script>

<style scoped>
   .nrNum {
     width: 100%;
     background-color: #f1f1f1;
     border: 0px #fff;
     color: #ff0000;
     font-size: 1.9em;
   }
  .REDnrNum {
    width: 100%;
    background-color: #f1f1f1;
    color: #ff0000;
    border: 0px #fff;
  }
  .ISPRIORITY {
    padding-left: 28%;
    padding-right: 28%;
    padding-bottom: 6%;
    padding-top: 6%;
    border: 2px solid #777;
    background-color: #cc0000;
    font-size: 1.25em;
    color: #ffffff;
  }
   .f1 {
     border: 1px solid #000000;
     padding: 2px;
   }
   /* initially hide client info ONLY WITHIN HEADER (not everywhere the component is used */
   .RequestInfoHeader .ClientInfo {
     display: none;
   }
 </style>
