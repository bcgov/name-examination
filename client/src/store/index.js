/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
