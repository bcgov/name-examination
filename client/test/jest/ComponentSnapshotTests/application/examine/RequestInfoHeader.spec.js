/* eslint-disable */
import { shallowMount } from '@vue/test-utils';
import RequestInfoHeader from '@/components/application/Examine/RequestInfoHeader';
import store from '@/store'

describe.skip('RequestInfoHeader.vue', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(RequestInfoHeader, {store: store});
    });

   it("renders a RequestInfoHeader component", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
