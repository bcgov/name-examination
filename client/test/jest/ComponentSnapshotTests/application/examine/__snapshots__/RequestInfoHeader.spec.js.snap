// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestInfoHeader.vue renders a RequestInfoHeader component 1`] = `
<v-flex-stub
  tag="div"
>
  <v-container-stub
    class=""
    fluid=""
    mt-2=""
    pa-0=""
    tag="div"
  >
    <v-layout-stub
      align-start=""
      pb-2=""
      row=""
      tag="div"
    >
      <v-flex-stub
        class="nr-number,text-left,mx-4,dk-grey"
        shrink=""
        tag="div"
      >
        <span
          id="nrNumberDisplay"
        >
          
        </span>
         
        <v-icon-stub>
          open_in_new
        </v-icon-stub>
      </v-flex-stub>
       
      <v-flex-stub
        grow="true"
        tag="div"
      >
        <v-layout-stub
          column=""
          tag="div"
        >
          <v-flex-stub
            fs-16=""
            fw-600=""
            id="request-type-desc"
            mt-2=""
            tag="div"
          >
            ERROR!!
          </v-flex-stub>
           
          <v-flex-stub
            fs-14=""
            fw-600=""
            tag="div"
          >
            
          </v-flex-stub>
        </v-layout-stub>
      </v-flex-stub>
       
      <actionbuttons-stub
        edit="function () { [native code] }"
        toggledetails="function () { [native code] }"
      />
    </v-layout-stub>
  </v-container-stub>
   
  <v-container-stub
    align-start=""
    class="info-banner-collapsed"
    fluid=""
    id="header-info-banner"
    justify-space-between=""
    ma-0=""
    py-0=""
    tag="div"
  >
    <v-layout-stub
      align-start=""
      justify-space-between=""
      pt-2=""
      tag="div"
    >
      <v-flex-stub
        fs-15=""
        ml-3=""
        shrink=""
        tag="div"
      >
        <v-layout-stub
          column=""
          tag="div"
        >
          <!---->
           
          <v-flex-stub
            id="div1"
            tag="div"
          >
            <b>
              Status:
            </b>
            <span
              id="nrStatusText"
            >
              
            
            </span>
          </v-flex-stub>
           
          <v-flex-stub
            tag="div"
          >
            <b>
              Examiner:
            </b>
             
          </v-flex-stub>
           
          <v-flex-stub
            tag="div"
          >
            <v-icon-stub
              class="mirrored"
              color="light-blue"
            >
              chat_bubble_outline
            </v-icon-stub>
             
            <b
              class="dark-blue--text"
            >
              
              0 C
              <span
                class="shortkey"
              >
                o
              </span>
              mments
            </b>
          </v-flex-stub>
        </v-layout-stub>
      </v-flex-stub>
       
      <v-flex-stub
        mx-2=""
        shrink=""
        tag="div"
      >
        <v-divider-stub
          vertical="true"
        />
      </v-flex-stub>
       
      <v-flex-stub
        fs-15=""
        tag="div"
      >
        <v-layout-stub
          column=""
          tag="div"
        >
          <v-flex-stub
            tag="div"
          >
            <b>
              Submit:
            </b>
             Invalid date
          </v-flex-stub>
           
          <!---->
           
          <!---->
           
          <v-flex-stub
            class="grey--text"
            tag="div"
          >
            <b>
              Expiry:
            </b>
             n/a
          </v-flex-stub>
           
          <v-flex-stub
            class="grey--text"
            tag="div"
          >
            <b>
              Consumed:
            </b>
             n/a
          
          </v-flex-stub>
           
          <!---->
        </v-layout-stub>
      </v-flex-stub>
       
      <v-flex-stub
        mx-2=""
        shrink=""
        tag="div"
      >
        <v-divider-stub
          vertical="true"
        />
      </v-flex-stub>
       
      <v-flex-stub
        fs-15=""
        tag="div"
      >
        <v-layout-stub
          column=""
          id="nwpta-ab"
          tag="div"
        >
          <nwpta-stub
            jurisdiction="AB"
            style="max-width: 120px;"
          />
        </v-layout-stub>
      </v-flex-stub>
       
      <v-flex-stub
        fs-15=""
        tag="div"
      >
        <v-layout-stub
          column=""
          id="nwpta-sk"
          tag="div"
        >
          <nwpta-stub
            jurisdiction="SK"
            style="max-width: 120px;"
          />
        </v-layout-stub>
      </v-flex-stub>
       
      <v-flex-stub
        mx-2=""
        shrink=""
        tag="div"
      >
        <v-divider-stub
          vertical="true"
        />
      </v-flex-stub>
       
      <infoheaderpopup-stub
        class=""
        infotype="information"
        save="function () { [native code] }"
        text=""
        title="Additional Info:"
      />
       
      <v-flex-stub
        mx-3=""
        shrink=""
        tag="div"
      >
        <v-divider-stub
          vertical="true"
        />
      </v-flex-stub>
       
      <infoheaderpopup-stub
        class=""
        infotype="nature"
        text=""
        title="N.O.B.:"
      >
        <v-layout-stub
          column=""
          tag="div"
        >
          <!---->
           
          <!---->
           
          <!---->
        </v-layout-stub>
      </infoheaderpopup-stub>
       
      <v-flex-stub
        mx-3=""
        shrink=""
        tag="div"
      >
        <v-divider-stub
          vertical="true"
        />
      </v-flex-stub>
       
      <infoheaderpopup-stub
        class="ma-0 pr-2"
        infotype="applicant"
        title="Applicant Information:"
      >
        <clientinfoheader-stub
          cancelsave="function () { [native code] }"
          save="function () { [native code] }"
        />
      </infoheaderpopup-stub>
    </v-layout-stub>
  </v-container-stub>
   
  <!---->
</v-flex-stub>
`;
